flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/nodes/exec/stream/GroupWindowAggregateJsonPlanTest_jsonplan/testProcTimeHopWindow.out

```diff
@@ -338,6 +343,12 @@
   }, {
     "id" : 7,
     "type" : "stream-exec-sink_1",
+    "configuration" : {
+      "table.exec.sink.keyed-shuffle" : "AUTO",
+      "table.exec.sink.not-null-enforcer" : "ERROR",
+      "table.exec.sink.type-length-enforcer" : "IGNORE",
```

Comment:
I forgot the discussion but why did we decide for this default? was it before the decision to disable legacy casting?

Reply:
I'll double check the original issue first.

Reply:
https://issues.apache.org/jira/browse/FLINK-24753
https://issues.apache.org/jira/browse/FLINK-24413

We didn't have an actual discussion on the sink constraint enforcer. I would argue that it's independent, and unless there is a Sink used that requires string compliance for `CHAR/VARCHAR/BINARY/VARBINARY` (like an RDBMS), we shouldn't enable it by default because of the performance overhead.

If users in their queries, use an explicit cast to those types, then by default (legacyCastBehaviour=false) will get the padding & trimming. If they use `CHAR/VARCHAR/BINARY/VARBINARY` as the target column type, without any explicit casting, then, if the target sink doesn't have strong compliance, everything will work out smoothly without performance penalty. If they need it they need to enable it on demand.

WDYT?

Reply:
Sounds good to me. I just wanted to rediscuss this one more time. But you are right. Now that the casting works correctly and we might fix more functions in the future, the enforcer runtime overhead might not be necessary.